src/scripts/clipperUI/clipper.tsx

```diff
@@ -620,9 +621,22 @@ class ClipperClass extends ComponentBase<ClipperState, {}> {
 
 		this.state.setState({ oneNoteApiResult: { status: Status.InProgress } });
 		SaveToOneNote.startClip(this.state).then((startClipPackage: StartClipPackage) => {
-			clipEvent.setCustomProperty(Log.PropertyName.Custom.CorrelationId, startClipPackage.responsePackage.request.getResponseHeader(Constants.HeaderValues.correlationId));
-			clipEvent.setCustomProperty(Log.PropertyName.Custom.AnnotationAdded, startClipPackage.annotationAdded);
-			this.state.setState({ oneNoteApiResult: { data: startClipPackage.responsePackage.parsedResponse, status: Status.Succeeded } });
+			RatingsHelper.incrementClipSuccessCount().then(() => {
+				// do nothing special on incrementClipSuccessCount resolve
+			}).then(() => {
+				// will happen regardless of success/failure of incrementClipSuccessCount
+				RatingsHelper.shouldShowRatingsPrompt(this.state).then((shouldShowRatingsPrompt) => {
```

Comment:
Do you know if there are other places in the code where decide behavior based on localStorage data? We front-load things like `userInfo`, which pulls from localStorage (e.g. `flightingInfo`). We could use a similar architecture for this. I think it would simplify the code by reducing the number of promises.